packages/types/src/codec/PairOf.ts

```diff
@@ -0,0 +1,36 @@
+// Copyright 2017-2018 @polkadot/types authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { Constructor, ConstructorDef } from '../types';
+import Base from './Base';
+import Tuple from './Tuple';
+
+// PairOf defines a general structure that creates a Tuple pair of a specific type. These values
+// are accessible via the first & second getters
+export default class PairOf<
+  T extends Base
+> extends Tuple<{ first: Constructor<T>, second: Constructor<T> }> {
+  static with<
+    S extends ConstructorDef = { [index: string]: Constructor<Base> }
+  > (Type: S): Constructor<Tuple<S>>;
+
+  static with<T extends Base> (Type: Constructor<T>): Constructor<PairOf<T>> {
```

Comment:
Not strictly overloading here. But as long as it's used internally I approve this.

Reply:
The issue with use comes in here - https://github.com/polkadot-js/api/pull/349/files#diff-a86daab9b4c7e3dcd4eb7d91f8feb064R143

(note the `any` cast, otherwise it complains bitterly)

Well, yes, best we can do for now - not overly excited about it either, but the lesser of evils.